### PR TITLE
Only set kubernetes server/port env vars on host-networked pods.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

In MKE clusters, the API server is only reachable via the MKE proxy from the host namespace because it is implemented as a DNS entry in /etc/hosts on the host.

## For PR author

- [x] Tests for change.

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Comment by tmjd** [comment]
This change seems to be very broad and will affect more than MKE clusters.
Specifically, I think there is a "minimal" setup that does not use Kubernetes services that this change would break since it is no longer setting the K8s env vars on several components.

Should this behavior be a special case when the provider is "DockerEnterprise" (which I think turned into MKE).

**Comment by fasaxc** [comment]
I was wary of making it specific to one environment since that makes the environments more different.  However, the "minimal" setup changes the calculus.  I'll condition on DockerEE.

**Comment by fasaxc** [comment]
@tmjd I've made it conditional on DockerEE and the actual value that is a special case.  I pushed the logic down into the shared code, thought that made it cleaner.

**Review by tmjd — APPROVED** [review]
I like that this also checks for the special `proxy.local`.

File: pkg/render/node.go
Comment (tmjd) [inline pkg/render/node.go:947]: I think you need to rebase on master. I think this will need to change to c.cfg.Installation.KubernetesProvider.

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
